pennylane_lightning_gpu/src/bindings/Bindings.cpp

```diff
@@ -85,6 +85,31 @@ void StateVectorCudaManaged_class_bindings(py::module &m) {
             return new StateVectorCudaManaged<PrecisionT>(
                 data_ptr, static_cast<std::size_t>(arr.size()));
         }))
+        .def(
+            "setBasisState",
+            [](StateVectorCudaManaged<PrecisionT> &sv, const size_t index,
+               const bool use_async) {
+                const std::complex<PrecisionT> value(1, 0);
+                sv.setBasisState(value, index, use_async);
+            },
+            "Create Basis State on GPU.")
+        .def(
+            "setStateVector",
+            [](StateVectorCudaManaged<PrecisionT> &sv,
+               const np_arr_sparse_ind &indices, const np_arr_c &state,
+               const bool use_async) {
+                using index_type = typename std::conditional<
+                    std::is_same<ParamT, float>::value, int32_t, int64_t>::type;
```

Comment:
Thanks @chaeyeunpark! As suggested by @mlxd , we decided to remove `state` for the moment. It will be great if you can have a test and let us know the performance. I think the answer is yes as I found "The input arrays x and y are automatically converted into the right types (they are of type numpy.dtype.int64 but need to be numpy.dtype.int32 and numpy.dtype.float32, respectively)." [here](https://github.com/pybind/pybind11/blob/d70f54b073d9961a4b9eebd151e721a1172ecada/docs/advanced/pycpp/numpy.rst)

Reply:
> Just a single question. I also note that we may need to readd `state` at some point if a user wants to have so. If we do not require any additional requirements, we may just allocate a memory handled by numpy (using capsule e.g. [pybind/pybind11#1042 (comment)](https://github.com/pybind/pybind11/issues/1042#issuecomment-325941022)) and copy data to there. I will also try to test your code using one of my script this afternoon and share a result.

We had considered this, but figured removing it altogether and requiring an explicit copy through the sync methods is currently the best way to guarantee no calls are made within PennyLane internal. We can make sure this is included in the docs so that an end-user can be made aware.

Reply:
Though, not against readding it in future with a warning, provided we are completely free of internal Python calling it anywhere as part of the execution pipeline.